zwave_js_server/model/driver.py

```diff
@@ -1,20 +1,41 @@
-"""Z-Wave JS Driver."""
+"""Provide a model for the Z-Wave JS Driver."""
+from enum import Enum
+
+from ..event import Event, EventBase
 from .controller import Controller
-from ..event import EventBase
 
 
 class Driver(EventBase):
+    """Represent a Z-Wave driver."""
+
     def __init__(self, state: dict):
-        super().__init__()
+        """Set up driver."""
+        super().__init__(EventType)
         self.controller = Controller(state)
 
-    def receive_event(self, event: dict):
+    def receive_event(self, event: Event) -> None:
         """Receive an event."""
-        if event["source"] != "driver":
+        if event.data["source"] != "driver":
             self.controller.receive_event(event)
             return
 
-        if event["event"] == "all nodes ready":
-            pass
+        self._handle_event_protocol(event)
+
+        self.emit(event.data["event"], event.data)
+
+    def handle_error(self, event: Event) -> None:
+        """Process a driver error event."""
+
+    def handle_driver_ready(self, event: Event) -> None:
+        """Process a driver ready event."""
```

Comment:
We don't forward these. The API only comes online once the driver is ready.

Reply:
Is it just this event ("driver ready") that we don't handle? We handle "error" and "all nodes ready"?

Reply:
`all nodes ready` we do. Error we don't forward because it's unclear to me when we get it. Is it during initialization or runtime too